lib/puppet/functions/json_data.rb

```diff
@@ -0,0 +1,18 @@
+# @since 4.8.0
+#
+Puppet::Functions.create_function(:json_data) do
+  dispatch :json_data do
+    param 'Struct[{path=>String[1]}]', :options
+    param 'Puppet::LookupContext', :context
+  end
```

Comment:
The yaml and json readers does not need options, but other functions that provide the entire data hash may need additional optional arguments.